src/Security/Domain/Authentication/AuthenticationService.php

```diff
@@ -0,0 +1,247 @@
+            }
+            $this->updateAuthenticationTokens($newAuthenticationTokens);
+        }
+
+        return true;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function createAuthenticationTokens(
+        string $sessionToken,
+        string $providerConfigurationName,
+        ContactInterface $contact,
+        ProviderToken $providerToken,
+        ?ProviderToken $providerRefreshToken
+    ): void {
+        $providerConfiguration = $this->providerService->findProviderConfigurationByConfigurationName(
+            $providerConfigurationName
+        );
+        if ($providerConfiguration === null || ($providerConfigurationId = $providerConfiguration->getId()) === null) {
+            throw ProviderServiceException::providerConfigurationNotFound($providerConfigurationName);
+        }
+        try {
+            $this->repository->addAuthenticationTokens(
+                $sessionToken,
+                $providerConfigurationId,
+                $contact->getId(),
+                $providerToken,
+                $providerRefreshToken
+            );
+        } catch (\Exception $ex) {
+            throw AuthenticationServiceException::addAuthenticationToken($ex);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function createAPIAuthenticationTokens(
+        string $token,
+        ProviderConfiguration $providerConfiguration,
+        ContactInterface $contact,
+        ProviderToken $providerToken,
+        ?ProviderToken $providerRefreshToken
+    ): void {
+        $this->debug('Creating authentication tokens for user', ['user' => $contact->getAlias()]);
+        try {
+            if ($providerConfiguration->getId() === null) {
+                throw new \InvalidArgumentException("Provider configuration can't be null");
+            }
+            $this->repository->addAPIAuthenticationTokens(
+                $token,
+                $providerConfiguration->getId(),
+                $contact->getId(),
+                $providerToken,
+                $providerRefreshToken
+            );
+        } catch (\Exception $ex) {
+            throw AuthenticationServiceException::addAuthenticationToken($ex);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function deleteSession(string $sessionToken): void
+    {
+        try {
+            $this->repository->deleteSession($sessionToken);
+        } catch (\Exception $ex) {
+            throw AuthenticationServiceException::deleteSession($ex);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function deleteExpiredSecurityTokens(): void
+    {
+        try {
+            $this->localProviderRepository->deleteExpiredSecurityTokens();
+        } catch (\Exception $ex) {
+            throw AuthenticationServiceException::deleteExpireToken($ex);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function hasValidSession(string $token, ProviderInterface $provider): bool
```

Comment:
This method seems no more used since: https://github.com/centreon/centreon/pull/9582/commits/4b1dd6b3c85c637ff2b45838e894fb5f8969c1af should we delete it ?